### PR TITLE
User can filter content based on location

### PR DESCRIPTION
**Pull request by raptorf1**
# [PT Story](https://www.pivotaltracker.com/story/show/166152505)

### Changes proposed in this PR
* User can filter articles in all categories and latest news section by country
* Updates all tests with missing mock data. Tests now passing with `rails server` down.
* Removes city selector from HeaderMain component

### What we learned
* Redux is our friend!
* Oliver and Greg are awesome!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,26 +17,29 @@ const requireSignIn = generateRequireSignInWrapper({
...
-          <Route exact path='/' render={() => (<Redirect to="/news" component={ArticlesByCategory} activeItem={'news'} />)}></Route>
-          <Route exact path={this.state.paths} component={ArticlesByCategory}></Route>
-          <Route exact path='/write-article' component={requireSignIn(WriteArticle)}></Route>
-          <Route exact path='/review-articles' component={requireSignIn(ListOfUnpublishedArticles)}></Route>
-          <Route exact path='/full-article' component={requireSignIn(FullArticle)}></Route>
-          <Route exact path='/login' component={LoginForm}></Route>
-          <Route exact path='/signup' component={SignUpForm}></Route>
-        </Switch>
+        <HeaderMain
+          handleChangeCountry={this.handleChangeCountry} />
```
Comment (oliverochman) [inline src/App.js]: remove this

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 2]
```diff
@@ -17,26 +17,29 @@ const requireSignIn = generateRequireSignInWrapper({
...
+          <Switch>
+            <Route exact path='/' render={() => (<Redirect to="/news" component={ArticlesByCategory} activeItem={'news'}/>)}></Route>
+            <Route exact path={this.state.paths} render={(props) => (<ArticlesByCategory {...props}/>)}></Route>
```
Comment (leiter007) [inline src/App.js]: Just out of curiosity: Is this line of passing props to ArticlesByCategory still 'necessary' with Redux?

**Review by leiter007 — COMMENTED** [review]
I think this looks amazing! Tried it out towards Heroku and with Cypress and works greatl! Really need to get some deeper insights into Redux I realized.. Left small clarification comment but nothing big.

File: src/App.js (hunk 1 above)
Comment (raptorf1) [reply src/App.js]: correct. removed.

File: src/App.js (hunk 2 above)
Comment (raptorf1) [reply src/App.js]: True also my friend. Not needed. Thanks.

**Pull request merged**